usl_pipeline/study_area_uploader/study_area_uploader/readers/polygon_readers.py

```diff
@@ -64,11 +86,25 @@ def read_polygons_from_shape_file(
                 )
 
         for fragment in fragments:
-            transformed_fragment = fragment
-            if transformer is not None:
-                xx, yy = transformer.transform(
-                    [p[0] for p in fragment], [p[1] for p in fragment]
+            if is_multi_polygon:
+                polygons.extend(
+                    [
```

Comment:
can avoid materializing this list by dropping the brackets -- `extend` will take a generator rather than a list
```python
polygons.extend(
    (
        geometry.Polygon(_transform_point_fragment(f, point_transformer)),
        mask_value,
    )
    for f in fragment
)
```